tools/tiled_ensemble/predictions/basic_joiner.py

```diff
@@ -0,0 +1,176 @@
+"""Concrete implementation of basic prediction joiner."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, List
+
+import torch
+from tools.tiled_ensemble.predictions.prediction_joiner import EnsemblePredictionJoiner
+from torch import Tensor
+
+
+class BasicPredictionJoiner(EnsemblePredictionJoiner):
+    """
+    Basic implementation of ensemble prediction joiner.
+
+    Tiles are put together and untiled.
+    Boxes are stacked per image.
+    Labels are combined with OR operator, meaning one anomalous tile -> anomalous image.
+    Scores are averaged.
+
+    Example:
+        >>> from tools.tiled_ensemble.ensemble_tiler import EnsembleTiler
+        >>> from tools.tiled_ensemble.predictions.prediction_data import BasicEnsemblePredictions
+        >>>
+        >>> tiler = EnsembleTiler(tile_size=256, stride=128, image_size=512)
+        >>> joiner = BasicPredictionJoiner(tiler)
+        >>> data = BasicEnsemblePredictions()
+        >>>
+        >>> # joiner needs to be setup with ensemble predictions storage object
+        >>> joiner.setup(data)
+        >>>
+        >>> # we can then start joining procedure for each batch
+        >>> joiner.join_tile_predictions(0)
+    """
+
+    def join_tiles(self, batch_data: dict, tile_key: str) -> Tensor:
+        """
+        Join tiles back into one tensor and perform untiling with tiler.
+
+        Args:
+            batch_data: Dictionary containing all tile predictions of current batch.
+            tile_key: Key used in prediction dictionary for tiles that we want to join
+
+        Returns:
+            Tensor of tiles in original (stitched) shape.
+        """
+        # tiles with index (0, 0) always exists
+        first_tiles = batch_data[(0, 0)][tile_key]
```

Comment:
is it first tiles? or first_tile

Reply:
It is first tiles, as there are more tiles in one batch. I can clarify this more in a comment I think.